components/admin_console/admin_console.jsx

```diff
@@ -73,63 +74,48 @@ export default class AdminConsole extends React.Component {
+        const schemas = sectionsConcat.map((item) => {
+            if (item.isHidden && item.isHidden(this.props.config, {}, this.props.license, this.props.buildEnterpriseReady)) {
+                return false;
+            }
+            if (!item.schema) {
+                return false;
+            }
+            return (
+                <Route
+                    key={item.url}
+                    path={`${this.props.match.url}/${item.url}`}
+                    render={(props) => (
+                        <SchemaAdminSettings
+                            {...extraProps}
+                            {...props}
+                            schema={item.schema}
                         />
-                    );
-                })}
-                {Object.values(AdminDefinition.settings).map(this.renderSectionRoutes.bind(this, extraProps))}
-                <Redirect to={`${this.props.match.url}/${firstUrl}`}/>
-            </Switch>
-        );
-    }
+                    )}
+                />
+            );
+        });
+        const defaultUrlSchema = sectionsConcat.find((item) => {
+            // return first available url
+            if (item.isHidden && item.isHidden(this.props.config, {}, this.props.license, this.props.buildEnterpriseReady)) {
+                return false;
+            }
+            if (item.schema) {
+                return item;
+            }
+            return false;
+        });
+        const defaultUrl = defaultUrlSchema.url;
 
-    renderSectionRoutes = (extraProps, section) => {
-        const firstUrl = Object.values(section).filter((i) => i.schema)[0].url;
         return (
-            <Route
-                key={section.url}
-                path={`${this.props.match.url}/${section.url}`}
-                render={(props) => (
-                    <Switch>
-                        {Object.values(section).map((item) => {
-                            if (!item.schema) {
-                                return null;
-                            }
-                            return (
-                                <Route
-                                    key={item.url}
-                                    path={`${props.match.url}/${item.url}`}
-                                    render={(subprops) => (
-                                        <SchemaAdminSettings
-                                            {...extraProps}
-                                            {...subprops}
-                                            schema={item.schema}
-                                        />
-                                    )}
-                                />
-                            );
-                        })}
-                        <Redirect to={`${props.match.url}/${firstUrl}`}/>
-                    </Switch>
-                )}
-            />
+            <Switch>
+                {schemas}
+                {<Redirect to={`${this.props.match.url}/${defaultUrl}`}/>}
```

Comment:
@bradjcoughlin would you know if the new behaviour is expected?

Also noticed the URL for some pages is `../admin_console/<feature>` instead of `../admin_console/<section>/<feature>`

Reply:
@jespino I've noticed this as well. I'm going to create a subsequent ticket to address some of the routing and URL concerns.

Reply:
@jasonblais @wiersgallak here is the follow-up ticket regarding URLs and routing: https://mattermost.atlassian.net/browse/MM-15027. I've noted that a decision needs to be made in regards to what should happen if a user browses to a URL that is hidden (in Environment for example), when the entire section is hidden. Should they be redirected to the first page in the system console?